src/components/button/button.ts

```diff
@@ -1,29 +1,34 @@
-import {Component, View, ViewEncapsulation, OnChanges, SimpleChange} from 'angular2/core';
-
+import {Component, View, ViewEncapsulation, OnChanges, SimpleChange, Input, Attribute, HostBinding, HostListener} from 'angular2/core';
 
 // TODO(jelbourn): Ink ripples.
 // TODO(jelbourn): Make the `isMouseDown` stuff done with one global listener.
+// TODO(kara): Convert attribute selectors to classes when attr maps become available
 
 @Component({
-  selector: '[md-button]:not(a), [md-raised-button]:not(a), [md-fab]:not(a)',
-  host: {
-    '(mousedown)': 'onMousedown()',
-    '(focus)': 'onFocus()',
-    '(blur)': 'onBlur()',
-    '[class.md-button-focus]': 'isKeyboardFocused',
-  },
+  selector: '[md-button]:not(a), [md-raised-button]:not(a), [md-fab]:not(a), [md-mini-fab]:not(a)',
   templateUrl: './components/button/button.html',
   styleUrls: ['./components/button/button.css'],
-  encapsulation: ViewEncapsulation.None,
+  encapsulation: ViewEncapsulation.None
 })
 export class MdButton {
-  /** Whether a mousedown has occured on this element in the last 100ms. */
+
+  /** Whether a mousedown has occurred on this element in the last 100ms. */
   isMouseDown: boolean = false;
 
+  @Input('color') color: string;
+
   /** Whether the button has focus from the keyboard (not the mouse). Used for class binding. */
-  isKeyboardFocused: boolean = false;
+  @HostBinding('class.md-button-focus') isKeyboardFocused: boolean = false;
 
-  onMousedown() {
+  @HostBinding('class')
+  get setClassList() {
```

Comment:
I'm not totally sure what this is doing, but it looks like it's just prefixing `md-` if this.color is one of those values.  Couldn't we do it simpler with something like: `return ['primary', 'accent', 'warn'].indexOf(this.color) >= 0 ? `md-${this.color}` : '';?